src/main/java/org/spongepowered/common/data/processor/data/entity/SneakingDataProcessor.java

```diff
@@ -0,0 +1,121 @@
+
+    @Override
+    public Optional<SneakingData> createFrom(DataHolder dataHolder) {
+        if (dataHolder instanceof Entity) {
+            final boolean sneaking = ((Entity) dataHolder).isSneaking();
+            return Optional.<SneakingData>of(new SpongeSneakingData(sneaking));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof Entity;
+    }
+
+    @Override
+    public Optional<SneakingData> from(DataHolder dataHolder) {
+        if (dataHolder instanceof Entity) {
+            final boolean sneaking = ((Entity) dataHolder).isSneaking();
+            return Optional.<SneakingData>of(new SpongeSneakingData(sneaking));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<SneakingData> fill(DataHolder dataHolder, SneakingData manipulator, MergeFunction overlap) {
+        if (dataHolder instanceof Entity) {
+            final Optional<SneakingData> oldData = from(dataHolder);
+            final SneakingData newData = checkNotNull(overlap, "Merge function was null!").merge(oldData.orNull(), manipulator);
+            final Value<Boolean> newValue = newData.sneaking();
+            return Optional.of(manipulator.set(newValue));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<SneakingData> fill(DataContainer container, SneakingData sneakingData) {
+        sneakingData.set(Keys.IS_SNEAKING, getData(container, Keys.IS_SNEAKING));
+        return Optional.of(sneakingData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, SneakingData manipulator, MergeFunction function) {
+        if (dataHolder instanceof Entity) {
+            final ImmutableValue<Boolean> newValue = manipulator.sneaking().asImmutable();
+            final SneakingData old = from(dataHolder).get();
+            final ImmutableValue<Boolean> oldValue = old.asImmutable().sneaking();
+            final SneakingData newData = checkNotNull(function, "function").merge(old, manipulator);
+            final boolean sneaking = newData.sneaking().get();
+            try {
+                ((Entity) dataHolder).setSneaking(sneaking);
+                return DataTransactionBuilder.successReplaceResult(newValue, oldValue);
+            } catch (Exception e) {
+                DataTransactionBuilder.errorResult(newValue);
+            }
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public Optional<ImmutableSneakingData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableSneakingData immutable) {
+        if (!key.equals(Keys.IS_SNEAKING)) {
+            return Optional.absent();
+        }
+        final ImmutableSneakingData data = ImmutableDataCachingUtil.getManipulator(ImmutableSneakingData.class, (Boolean) value);
```

Comment:
You need to provide the implemented `ImmutableSpongeSneakingData` class to construct it.